src/auth.js

```diff
@@ -67,7 +67,7 @@ const storeApiKey = (key) => {
 }
 
 const getStateObject = (key) => {
-  let state = key // The key here is the state value from auth url.
+  let state = JSON.stringify(key) // The key here is the state value from auth url.
```

Comment:
This feels odd. If the goal is to normalize the data, stringifying the plain text seems more onerous than parsing only the stringified text. How about this?
```
let state = key
if (window.localStorage) {
    const stringifiedStage = window.localStorage.getItem(key)
    state = stringifiedState ? JSON.parse(stringifiedState) : state
}
return state
```